Test/civl/rwlock.bpl

```diff
@@ -0,0 +1,152 @@
+// RUN: %parallel-boogie "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+// In this example we show how to model a reader-writer lock.
+
+type {:linear "tid"} Tid;
+
+type {:datatype} OptionTid;
```

Comment:
You might consider moving this definition to library.bpl since it seems to be often used in Civl examples.

Reply:
Also, make it generic if you do.

Reply:
Made it generic. Let's do a move to library separately.